pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -520,39 +519,44 @@ func (r *ReconcileNetworkAddonsConfig) stopTrackingObjects() {
 	r.statusManager.SetFromPods()
 }
 
-func updateObjectsLabels(crLables map[string]string, objs []*unstructured.Unstructured) error {
+func updateObjectsLabels(crLabels map[string]string, objs []*unstructured.Unstructured) error {
 	var err error
 	for _, obj := range objs {
 		labels := obj.GetLabels()
 		if labels == nil {
 			labels = map[string]string{}
 		}
-		// Label objects with version of the operator they were created by
-		labels[cnaov1.SchemeGroupVersion.Group+"/version"] = operatorVersionLabel
-		labels[names.PROMETHEUS_LABEL_KEY] = names.PROMETHEUS_LABEL_VALUE
-		labels[names.MANAGED_BY_LABEL_KEY] = names.MANAGED_BY_LABEL_DEFAULT_VALUE
+		if !isOperatorNamespace(obj) {
+			// Label objects with version of the operator they were created by
+			labels[cnaov1.SchemeGroupVersion.Group+"/version"] = operatorVersionLabel
+			labels[names.PROMETHEUS_LABEL_KEY] = names.PROMETHEUS_LABEL_VALUE
+			labels[names.MANAGED_BY_LABEL_KEY] = names.MANAGED_BY_LABEL_DEFAULT_VALUE
+
+			appLabelKeys := []string{names.COMPONENT_LABEL_KEY, names.PART_OF_LABEL_KEY, names.VERSION_LABEL_KEY}
+			labels = updateLabelsFromCR(labels, crLabels, appLabelKeys)
+			if err != nil {
+				return err
+			}
 
-		appLabelKeys := []string{names.COMPONENT_LABEL_KEY, names.PART_OF_LABEL_KEY, names.VERSION_LABEL_KEY}
-		labels = updateLabelsFromCR(labels, crLables, appLabelKeys)
-		if err != nil {
-			return err
+			templateLabelKeys := append(appLabelKeys, names.PROMETHEUS_LABEL_KEY, names.MANAGED_BY_LABEL_KEY)
+			err = updateObjectTemplateLabels(obj, labels, templateLabelKeys)
+			if err != nil {
+				return err
+			}
+		} else {
+			delete(labels, names.KUBEMACPOOL_CONTROL_PLANE_KEY)
```

Comment:
if we remove this label from kmp
https://github.com/k8snetworkplumbingwg/kubemacpool/blob/e5c69e0641317336e13405c5476c7fb750e32ff4/config/default/manager/manager.yaml#L5

we can remove this line of code (but will still need to remove it at `cleanUpNamespaceLabels` for backward compability

Reply:
I think that removing it from KMP directly would be the best option. In general, I think that operators should not manager the Namespace they are deployed into